odoo/fields.py

```diff
@@ -1154,9 +1153,22 @@ class Integer(Field):
     """ Encapsulates an :class:`int`. """
     type = 'integer'
     column_type = ('int4', 'int4')
+    bigint = False
 
     group_operator = 'sum'
 
+    def __init__(self, string=Default, **kwargs):
+        super(Integer, self).__init__(string=string, **kwargs)
+        if kwargs.get('bigint'):
+            self.column_type = ('int8', 'int8')
+            self.column_cast_from = ('int4',)
+
+    def update_db_column(self, model, column):
+        """ Advertize column type as int8 if that's the current type """
+        if column and column['udt_name'] == 'int8':
+            self.column_type = ('int8', 'int8')
```

Comment:
If you go for the alternative above, then this should be
```suggestion
            self.bigint = True
```

Reply:
Taken